### ISSUE TITLE
Add support for DNS over HTTPS (DoH)

### ISSUE THREAD

**Issue by jportner** [DesktopECHO/Pi-Hole-for-WSL1#12]
For the privacy-conscious, it would be nice to use this in conjunction with DNS over HTTPS. It looks like the `cloudflared` binary provides support for this which is easy to integrate with pihole:
* https://docs.pi-hole.net/guides/dns/cloudflared/
* https://nathancatania.com/posts/pihole-dns-doh/

**Comment by DesktopECHO** [comment]
This is a great idea, I'll see if this can be integrated when I get some free time.
Thanks,
D.

**Comment by jportner** [comment]
Great, glad you agree! If I get a chance before you do, I'll submit a PR 😄

**Comment by lordcyb3r** [comment]
frickin bump my dude. would love to see that!

**Comment by lordcyb3r** [comment]
out of curiosity though did you attempt to install in via apt?

**Comment by jportner** [comment]
> out of curiosity though did you attempt to install in via apt?

No, I didn't attempt it yet 🙂

**Comment by lordcyb3r** [comment]
> > out of curiosity though did you attempt to install in via apt?
> 
> No, I didn't attempt it yet 🙂

well, you are in luck my friend! I got it working.

**Comment by lordcyb3r** [comment]
make sure to install systemctl. Than reference this to launch with windows (pi-hole launcher.cmd):

@WSLCONFIG /T Pi-hole                                                                                                               
@ECHO [Pi-Hole Launcher]                                                                                                          
@"C:\Program Files\Pi-hole\LxRunOffline.exe" r -n Pi-hole -c  "apt-get -qq remove dhcpcd5 > /dev/null 2>&1"  
@"C:\Program Files\Pi-hole\LxRunOffline.exe" r -n Pi-hole -c  "systemctl start cloudflared"        
                                                  
@"C:\Program Files\Pi-hole\LxRunOffline.exe" r -n Pi-hole -c  "for rc_service in /etc/rc2.d/S*; do [[ -e $rc_service ]] && $rc_service start ; done ; sleep 3"                             
@EXIT

**Comment by lordcyb3r** [comment]
TLDR:
@"C:\[install path]\LxRunOffline.exe" r -n Pi-hole -c "systemctl start cloudflared"  -> pi-hole launcher.cmd

**Comment by lordcyb3r** [comment]
> This is a great idea, I'll see if this can be integrated when I get some free time.
> Thanks,
> D.

made your job easier ^^^^

**Comment by lordcyb3r** [comment]
no worries dawg, happy to do it. FYI seems like you uploaded an empty .cmd file this time XD

**Comment by lordcyb3r** [comment]
I was just surprised that it worked without a hitch. good news for us I guess

**Comment by DesktopECHO** [comment]
Hi there, I ran into an issue and had to go AFK. I will correct this when I
get back to my desk in a couple of hours.
Cheers,
D.
-- 
Daniel Milisic - DesktopECHO Inc.
3865 Lake Shore Blvd W, Suite 307
Toronto, ON - M8W 0A2
(416) 899-7826 - Toronto
(519) 729-0111 - Waterloo

**Comment by lordcyb3r** [comment]
its all good dawg. just let me know if you need anything. ill have some time later today if you need a lab rat XD

**Comment by DesktopECHO** [comment]
Updated the script, looks solid now!
